core/src/jvmMain/kotlin/com/ustadmobile/core/catalog/contenttype/ApacheIndexerPlugin.kt

```diff
@@ -0,0 +1,163 @@
+            if (table.isEmpty()) {
+                return@withContext null
+            }
+
+            val altElement = table.select("[alt]")
+            if (altElement.isNullOrEmpty() || altElement.attr("alt") != "[ICO]") {
+                return@withContext null
+            }
+
+            val entry = ContentEntryWithLanguage()
+            entry.title = document.title().substringAfterLast("/")
+            entry.sourceUrl =  uri.uri.toString()
+            entry.leaf = false
+            entry.contentTypeFlag = ContentEntry.TYPE_COLLECTION
+
+            return@withContext MetadataResult(entry, PLUGIN_ID)
+        }
+    }
+
+    override suspend fun processJob(jobItem: ContentJobItemAndContentJob, process: ProcessContext, progress: ContentJobProgressListener): ProcessResult {
+        val contentJobItem = jobItem.contentJobItem ?: throw IllegalArgumentException("missing job item")
+        val jobUri = contentJobItem.sourceUri ?: return ProcessResult(JobStatus.FAILED)
+        withContext(Dispatchers.Default) {
+            val uri = DoorUri.parse(jobUri)
+            val localUri = process.getLocalUri(uri, context, di)
+            val data: InputStream? = localUri.openInputStream(context)
+            val document = Jsoup.parse(data, "UTF-8", uri.uri.toURL().toString())
+
+            val apacheDir = createTemporaryDir("apache-${jobItem.contentJobItem?.cjiUid}")
+
+            document.select("tr:has([alt])").forEachIndexed { counter, it ->
+
+                val alt = it.select("td [alt]").attr("alt")
+                try {
+                    val element = it.select("td a")
+                    val href = element.attr("href")
+                    if(href.isEmpty()){
+                        return@forEachIndexed
+                    }
+                    val hrefUrl = URL(uri.uri.toURL(), href)
+
+                    if(alt == "[PARENTDIR]" || alt == "[ICO]" || alt == ""){
+                        return@forEachIndexed
+                    }
+
+                    if (alt == "[DIR]") {
+
+                        ContentJobItem().apply {
+                            cjiJobUid = contentJobItem.cjiJobUid
+                            sourceUri = hrefUrl.toURI().toString()
+                            cjiItemTotal = sourceUri?.let { DoorUri.parse(it).getSize(context, di)  } ?: 0L
+                            cjiPluginId = PLUGIN_ID
+                            cjiContentEntryUid = 0
+                            cjiIsLeaf = false
+                            cjiParentContentEntryUid = contentJobItem.cjiContentEntryUid
+                            cjiParentCjiUid = contentJobItem.cjiUid
+                            cjiConnectivityNeeded = false
+                            cjiStatus = JobStatus.QUEUED
+                            cjiUid = db.contentJobItemDao.insertJobItem(this)
+                        }
+
+                    } else {
+
+                        val processContext = ProcessContext(apacheDir, mutableMapOf())
```

Comment:
Doing this will lead to the resource being downloaded twice, it will be inefficient. Right now these temporary directories are not being cleaned up. 

I think the Indexer should not try to extract metadata. Just check the mime type. If the mime type is unknown, or the mime type matches a supported format, then create a ContentJobItem for it. Then the ContentJobItem itself will download it (once), look at it, and process it (or, ignore it if it can't be processed).